utils/list-licenses/list-licenses.sh

```diff
@@ -99,4 +99,70 @@ do
 done
 
 # Special care for Rust
-find "${LIBS_PATH}/rust_vendor/" -name 'Cargo.toml' | xargs ${GREP_CMD} 'license = "' | (${GREP_CMD} -v -P 'MIT|Apache|MPL|ISC|BSD|Unicode|Zlib|CC0-1.0|CDLA' && echo "Fatal error: unrecognized licenses in the Rust code" >&2 && exit 1 || true)
+for dependency in $(find "${LIBS_PATH}/rust_vendor/" -name 'Cargo.toml');
+do
+    FOLDER=$(dirname "$dependency")
+    NAME=$(echo "$dependency" | awk -F'/' '{ print $(NF-1) }' | awk -F'-' '{ NF=(NF-1); print $0 }')
+    LICENSE_TYPE=$(${GREP_CMD} 'license = "' "$dependency"  | cut -d '"' -f2)
+    if echo "${LICENSE_TYPE}" | ${GREP_CMD} -v -P 'MIT|Apache|MPL|ISC|BSD|Unicode|Zlib|CC0-1.0|CDLA-Permissive';
+    then
+        echo "Fatal error: unrecognized licenses ($LICENSE_TYPE) in the Rust code"
+        exit 1
+    fi
+
+    LICENSE_PATH=""
+    declare -a arr=(
+      "LICENSE"
+      "LICENCE"
+      "LICENSE.md"
+      "LICENSE.txt"
+      "LICENSE.TXT"
+      "COPYING"
+      "LICENSE_APACHE"
+      "LICENSE-APACHE"
+      "license-apache-2.0"
+      "LICENSES/Apache-2.0.txt"
+      "LICENSE-MIT"
+      "LICENSE-MIT.txt"
+      "LICENSE-MIT.md"
+      "LICENSE.MIT"
+    )
+    for possible_path in "${arr[@]}"
+    do
+        if test -f "${FOLDER}/${possible_path}";
+        then
+            LICENSE_PATH="${FOLDER}/${possible_path}"
+            break
+        fi
+    done
+
+    if [ -z "${LICENSE_PATH}" ];
+    then
+        # It's important that we match exactly the license that we want for those projects that don't include a LICENSE file
+        if [ "$LICENSE_TYPE" == "Apache-2.0" ] ||
```

Comment:
Looks like many crates have little variations of "Apache 2.0" as license. Maybe its enough to do a regexp search for "Apach-2.0" instead.

Reply:
I don't like regexp because it might match odd things somebody wrote there. I would rather review each case individually to confirm we can use Apache 2.0 for that crate.